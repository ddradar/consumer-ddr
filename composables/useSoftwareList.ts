import type { Ref } from 'vue'

import type { Software } from '~~/src/content'

const keys = [
  'slug',
  'title',
  'color',
  'platform',
  'region',
  'launched'
] as const
export type SoftwareListData = Pick<Software, typeof keys[number]>

export default async function () {
  const { data } = await useAsyncData('/software', () =>
    queryContent<Software>('series')
      .sort({ launched: 1 })
      .only([...keys])
      .find()
  )

  return { softwareList: data as Ref<SoftwareListData[]> }
}
