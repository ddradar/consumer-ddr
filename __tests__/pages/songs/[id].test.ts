import { RouterLinkStub } from '@vue/test-utils'
import { beforeAll, describe, expect, test, vi } from 'vitest'
import { ref } from 'vue'

import useSoftwareList from '~~/composables/useSoftwareList'
import first from '~~/content/1st-jp/trip-machine.json'
import third from '~~/content/3rd/trip-machine.json'
import SongDetail from '~~/pages/songs/[id].vue'

import { mockSoftwareList, mountAsync, plugins } from '../../test-utils'

vi.mock('~~/composables/useSoftwareList')

const thirdSeries = {
  slug: '3rd',
  color: 'crimson',
  title: 'Dance Dance Revolution 3rd MIX',
  platform: 'Play Station',
  region: 'JP',
  launched: '2000-06-01'
} as const
const rawSongData = [first, third].map((d) => ({
  series: d.series,
  name: d.name,
  artist: d.artist,
  bpm: d.bpm,
  charts: d.charts
}))

describe('pages/songs/[id].vue', () => {
  const song = {
    name: first.name,
    artist: first.artist,
    bpm: first.bpm,
    series: [first.series, third.series],
    charts: [
      {
        playStyle: 'SINGLE',
        difficulty: 1,
        seriesList: [
          {
            slug: mockSoftwareList[0].slug,
            title: mockSoftwareList[0].title,
            color: mockSoftwareList[0].color,
            level: 6
          },
          {
            slug: thirdSeries.slug,
            title: thirdSeries.title,
            color: thirdSeries.color,
            level: 6
          }
        ]
      },
      {
        playStyle: 'SINGLE',
        difficulty: 2,
        seriesList: [
          {
            slug: mockSoftwareList[0].slug,
            title: mockSoftwareList[0].title,
            color: mockSoftwareList[0].color,
            level: 7
          },
          {
            slug: thirdSeries.slug,
            title: thirdSeries.title,
            color: thirdSeries.color,
            level: 7
          }
        ]
      },
      {
        playStyle: 'SINGLE',
        difficulty: 3,
        seriesList: [
          {
            slug: mockSoftwareList[0].slug,
            title: mockSoftwareList[0].title,
            color: mockSoftwareList[0].color,
            level: 8
          },
          {
            slug: thirdSeries.slug,
            title: thirdSeries.title,
            color: thirdSeries.color,
            level: 8
          }
        ]
      },
      {
        playStyle: 'DOUBLE',
        difficulty: 1,
        seriesList: [
          {
            slug: mockSoftwareList[0].slug,
            title: mockSoftwareList[0].title,
            color: mockSoftwareList[0].color,
            level: 7
          },
          {
            slug: thirdSeries.slug,
            title: thirdSeries.title,
            color: thirdSeries.color,
            level: 7
          }
        ]
      },
      {
        playStyle: 'DOUBLE',
        difficulty: 2,
        seriesList: [
          {
            slug: mockSoftwareList[0].slug,
            title: mockSoftwareList[0].title,
            color: mockSoftwareList[0].color,
            level: 8
          },
          {
            slug: thirdSeries.slug,
            title: thirdSeries.title,
            color: thirdSeries.color,
            level: 8
          }
        ]
      },
      {
        playStyle: 'COUPLE',
        difficulty: 1,
        seriesList: [
          {
            slug: mockSoftwareList[0].slug,
            title: mockSoftwareList[0].title,
            color: mockSoftwareList[0].color,
            level: 6
          }
        ]
      },
      {
        playStyle: 'COUPLE',
        difficulty: 2,
        seriesList: [
          {
            slug: mockSoftwareList[0].slug,
            title: mockSoftwareList[0].title,
            color: mockSoftwareList[0].color,
            level: 7
          }
        ]
      },
      {
        playStyle: 'COUPLE',
        difficulty: 3,
        seriesList: [
          {
            slug: mockSoftwareList[0].slug,
            title: mockSoftwareList[0].title,
            color: mockSoftwareList[0].color,
            level: 8
          }
        ]
      },
      {
        playStyle: 'UNISON',
        difficulty: 1,
        seriesList: [
          {
            slug: thirdSeries.slug,
            title: thirdSeries.title,
            color: thirdSeries.color,
            level: 6
          }
        ]
      },
      {
        playStyle: 'UNISON',
        difficulty: 2,
        seriesList: [
          {
            slug: thirdSeries.slug,
            title: thirdSeries.title,
            color: thirdSeries.color,
            level: 7
          }
        ]
      },
      {
        playStyle: 'UNISON',
        difficulty: 3,
        seriesList: [
          {
            slug: thirdSeries.slug,
            title: thirdSeries.title,
            color: thirdSeries.color,
            level: 8
          }
        ]
      }
    ]
  }
  const stubs = { NuxtLink: RouterLinkStub }
  const global = { plugins, stubs }

  beforeAll(() => {
    vi.mocked(resolveComponent).mockReturnValue(RouterLinkStub as any)
    vi.mocked(useRoute).mockReturnValue({
      params: { id: 'trip-machine' }
    } as any)
    const softwareList = ref([...mockSoftwareList, thirdSeries])
    vi.mocked(useSoftwareList).mockResolvedValue({ softwareList })
    vi.mocked(useAsyncData).mockReset()
    vi.mocked(useAsyncData).mockResolvedValue({ data: ref(song) } as any)
  })

  describe('snapshot test', () => {
    test('renders correctly', async () => {
      const wrapper = await mountAsync(SongDetail, { global })
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('useAsyncData()', () => {
    test('calls with transform option', async () => {
      // Arrange - Act
      await mountAsync(SongDetail, { global })
      const transform = vi.mocked(useAsyncData).mock.calls[0][2]!.transform!

      // Assert
      expect(transform(rawSongData)).toStrictEqual(song)
    })
  })
})
