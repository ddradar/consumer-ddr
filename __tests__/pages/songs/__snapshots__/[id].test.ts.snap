// Vitest Snapshot v1

exports[`pages/songs/[id].vue > snapshot test > renders correctly 1`] = `
<section
  class="section"
>
  <h1
    class="title"
  >
    TRIP MACHINE / DE-SIRE
  </h1>
  <span
    class="subtitle"
  >
    BPM 160
  </span>
  <section
    class="content"
  >
    <ul>
      
      <li>
        <a>
          Dance Dance Revolution
        </a>
      </li>
      <li>
        <a>
          Dance Dance Revolution 3rd MIX
        </a>
      </li>
      
    </ul>
  </section>
  <div
    class="b-table"
  >
    <div
      style="display: none;"
    >
      
      <span
        data-id="1"
      >
        PlayStyle
      </span>
      <span
        data-id="2"
      >
        Difficulty
      </span>
      <span
        data-id="3"
      >
        Level
      </span>
      
    </div>
    <!--v-if-->
    <!--v-if-->
    <div
      class="table-wrapper"
    >
      <table
        class="table is-striped"
        tabindex="false"
      >
        <!--v-if-->
        <thead>
          <tr>
            <!--v-if-->
            <!--v-if-->
            
            <th
              class=""
              draggable="false"
            >
              <span>
                PlayStyle 
                <span
                  class="th-wrap sort-icon"
                  style="display: none;"
                >
                  <span
                    class="icon is-small"
                  >
                    <i
                      class="mdi mdi-arrow-up"
                    />
                  </span>
                </span>
              </span>
            </th>
            <th
              class=""
              draggable="false"
            >
              <span>
                Difficulty 
                <span
                  class="th-wrap sort-icon"
                  style="display: none;"
                >
                  <span
                    class="icon is-small"
                  >
                    <i
                      class="mdi mdi-arrow-up"
                    />
                  </span>
                </span>
              </span>
            </th>
            <th
              class=""
              draggable="false"
            >
              <span>
                Level 
                <span
                  class="th-wrap sort-icon"
                  style="display: none;"
                >
                  <span
                    class="icon is-small"
                  >
                    <i
                      class="mdi mdi-arrow-up"
                    />
                  </span>
                </span>
              </span>
            </th>
            
            <!--v-if-->
          </tr>
          <!--v-if-->
          <!--v-if-->
        </thead>
        <tbody>
          
          
          <tr
            class=""
            draggable="false"
          >
            <!--v-if-->
            <!--v-if-->
            
            <td
              class=""
              data-label="PlayStyle"
              scoped=""
            >
              SINGLE
            </td>
            <td
              class=""
              data-label="Difficulty"
              scoped=""
            >
              1
            </td>
            <td
              class=""
              data-label="Level"
              scoped=""
            >
              <div
                class="field is-grouped is-grouped-multiline"
              >
                
                <div
                  class="control"
                >
                  <div
                    class="b-tooltip is-primary is-top"
                  >
                    <transition-stub>
                      <div
                        class="tooltip-content is-top"
                        style="display: none;"
                      >
                        <span
                          class=""
                        />
                        
                        Dance Dance Revolution
                        
                      </div>
                    </transition-stub>
                    <div
                      class="tooltip-trigger"
                    >
                      
                      <div
                        class="tags has-addons"
                      >
                        <span
                          class="is-series tag"
                          data-v-c1fd21ef=""
                        >
                          
                          1st-jp
                          
                        </span>
                        <span
                          class="tag"
                        >
                          6
                        </span>
                      </div>
                      
                    </div>
                  </div>
                </div>
                <div
                  class="control"
                >
                  <div
                    class="b-tooltip is-primary is-top"
                  >
                    <transition-stub>
                      <div
                        class="tooltip-content is-top"
                        style="display: none;"
                      >
                        <span
                          class=""
                        />
                        
                        Dance Dance Revolution 3rd MIX
                        
                      </div>
                    </transition-stub>
                    <div
                      class="tooltip-trigger"
                    >
                      
                      <div
                        class="tags has-addons"
                      >
                        <span
                          class="is-series tag"
                          data-v-c1fd21ef=""
                        >
                          
                          3rd
                          
                        </span>
                        <span
                          class="tag"
                        >
                          6
                        </span>
                      </div>
                      
                    </div>
                  </div>
                </div>
                
              </div>
            </td>
            
            <!--v-if-->
          </tr>
          <transition-stub>
            <!--v-if-->
          </transition-stub>
          <!--v-if-->
          
          
          <tr
            class=""
            draggable="false"
          >
            <!--v-if-->
            <!--v-if-->
            
            <td
              class=""
              data-label="PlayStyle"
              scoped=""
            >
              SINGLE
            </td>
            <td
              class=""
              data-label="Difficulty"
              scoped=""
            >
              2
            </td>
            <td
              class=""
              data-label="Level"
              scoped=""
            >
              <div
                class="field is-grouped is-grouped-multiline"
              >
                
                <div
                  class="control"
                >
                  <div
                    class="b-tooltip is-primary is-top"
                  >
                    <transition-stub>
                      <div
                        class="tooltip-content is-top"
                        style="display: none;"
                      >
                        <span
                          class=""
                        />
                        
                        Dance Dance Revolution
                        
                      </div>
                    </transition-stub>
                    <div
                      class="tooltip-trigger"
                    >
                      
                      <div
                        class="tags has-addons"
                      >
                        <span
                          class="is-series tag"
                          data-v-c1fd21ef=""
                        >
                          
                          1st-jp
                          
                        </span>
                        <span
                          class="tag"
                        >
                          7
                        </span>
                      </div>
                      
                    </div>
                  </div>
                </div>
                <div
                  class="control"
                >
                  <div
                    class="b-tooltip is-primary is-top"
                  >
                    <transition-stub>
                      <div
                        class="tooltip-content is-top"
                        style="display: none;"
                      >
                        <span
                          class=""
                        />
                        
                        Dance Dance Revolution 3rd MIX
                        
                      </div>
                    </transition-stub>
                    <div
                      class="tooltip-trigger"
                    >
                      
                      <div
                        class="tags has-addons"
                      >
                        <span
                          class="is-series tag"
                          data-v-c1fd21ef=""
                        >
                          
                          3rd
                          
                        </span>
                        <span
                          class="tag"
                        >
                          7
                        </span>
                      </div>
                      
                    </div>
                  </div>
                </div>
                
              </div>
            </td>
            
            <!--v-if-->
          </tr>
          <transition-stub>
            <!--v-if-->
          </transition-stub>
          <!--v-if-->
          
          
          <tr
            class=""
            draggable="false"
          >
            <!--v-if-->
            <!--v-if-->
            
            <td
              class=""
              data-label="PlayStyle"
              scoped=""
            >
              SINGLE
            </td>
            <td
              class=""
              data-label="Difficulty"
              scoped=""
            >
              3
            </td>
            <td
              class=""
              data-label="Level"
              scoped=""
            >
              <div
                class="field is-grouped is-grouped-multiline"
              >
                
                <div
                  class="control"
                >
                  <div
                    class="b-tooltip is-primary is-top"
                  >
                    <transition-stub>
                      <div
                        class="tooltip-content is-top"
                        style="display: none;"
                      >
                        <span
                          class=""
                        />
                        
                        Dance Dance Revolution
                        
                      </div>
                    </transition-stub>
                    <div
                      class="tooltip-trigger"
                    >
                      
                      <div
                        class="tags has-addons"
                      >
                        <span
                          class="is-series tag"
                          data-v-c1fd21ef=""
                        >
                          
                          1st-jp
                          
                        </span>
                        <span
                          class="tag"
                        >
                          8
                        </span>
                      </div>
                      
                    </div>
                  </div>
                </div>
                <div
                  class="control"
                >
                  <div
                    class="b-tooltip is-primary is-top"
                  >
                    <transition-stub>
                      <div
                        class="tooltip-content is-top"
                        style="display: none;"
                      >
                        <span
                          class=""
                        />
                        
                        Dance Dance Revolution 3rd MIX
                        
                      </div>
                    </transition-stub>
                    <div
                      class="tooltip-trigger"
                    >
                      
                      <div
                        class="tags has-addons"
                      >
                        <span
                          class="is-series tag"
                          data-v-c1fd21ef=""
                        >
                          
                          3rd
                          
                        </span>
                        <span
                          class="tag"
                        >
                          8
                        </span>
                      </div>
                      
                    </div>
                  </div>
                </div>
                
              </div>
            </td>
            
            <!--v-if-->
          </tr>
          <transition-stub>
            <!--v-if-->
          </transition-stub>
          <!--v-if-->
          
          
          <tr
            class=""
            draggable="false"
          >
            <!--v-if-->
            <!--v-if-->
            
            <td
              class=""
              data-label="PlayStyle"
              scoped=""
            >
              DOUBLE
            </td>
            <td
              class=""
              data-label="Difficulty"
              scoped=""
            >
              1
            </td>
            <td
              class=""
              data-label="Level"
              scoped=""
            >
              <div
                class="field is-grouped is-grouped-multiline"
              >
                
                <div
                  class="control"
                >
                  <div
                    class="b-tooltip is-primary is-top"
                  >
                    <transition-stub>
                      <div
                        class="tooltip-content is-top"
                        style="display: none;"
                      >
                        <span
                          class=""
                        />
                        
                        Dance Dance Revolution
                        
                      </div>
                    </transition-stub>
                    <div
                      class="tooltip-trigger"
                    >
                      
                      <div
                        class="tags has-addons"
                      >
                        <span
                          class="is-series tag"
                          data-v-c1fd21ef=""
                        >
                          
                          1st-jp
                          
                        </span>
                        <span
                          class="tag"
                        >
                          7
                        </span>
                      </div>
                      
                    </div>
                  </div>
                </div>
                <div
                  class="control"
                >
                  <div
                    class="b-tooltip is-primary is-top"
                  >
                    <transition-stub>
                      <div
                        class="tooltip-content is-top"
                        style="display: none;"
                      >
                        <span
                          class=""
                        />
                        
                        Dance Dance Revolution 3rd MIX
                        
                      </div>
                    </transition-stub>
                    <div
                      class="tooltip-trigger"
                    >
                      
                      <div
                        class="tags has-addons"
                      >
                        <span
                          class="is-series tag"
                          data-v-c1fd21ef=""
                        >
                          
                          3rd
                          
                        </span>
                        <span
                          class="tag"
                        >
                          7
                        </span>
                      </div>
                      
                    </div>
                  </div>
                </div>
                
              </div>
            </td>
            
            <!--v-if-->
          </tr>
          <transition-stub>
            <!--v-if-->
          </transition-stub>
          <!--v-if-->
          
          
          <tr
            class=""
            draggable="false"
          >
            <!--v-if-->
            <!--v-if-->
            
            <td
              class=""
              data-label="PlayStyle"
              scoped=""
            >
              DOUBLE
            </td>
            <td
              class=""
              data-label="Difficulty"
              scoped=""
            >
              2
            </td>
            <td
              class=""
              data-label="Level"
              scoped=""
            >
              <div
                class="field is-grouped is-grouped-multiline"
              >
                
                <div
                  class="control"
                >
                  <div
                    class="b-tooltip is-primary is-top"
                  >
                    <transition-stub>
                      <div
                        class="tooltip-content is-top"
                        style="display: none;"
                      >
                        <span
                          class=""
                        />
                        
                        Dance Dance Revolution
                        
                      </div>
                    </transition-stub>
                    <div
                      class="tooltip-trigger"
                    >
                      
                      <div
                        class="tags has-addons"
                      >
                        <span
                          class="is-series tag"
                          data-v-c1fd21ef=""
                        >
                          
                          1st-jp
                          
                        </span>
                        <span
                          class="tag"
                        >
                          8
                        </span>
                      </div>
                      
                    </div>
                  </div>
                </div>
                <div
                  class="control"
                >
                  <div
                    class="b-tooltip is-primary is-top"
                  >
                    <transition-stub>
                      <div
                        class="tooltip-content is-top"
                        style="display: none;"
                      >
                        <span
                          class=""
                        />
                        
                        Dance Dance Revolution 3rd MIX
                        
                      </div>
                    </transition-stub>
                    <div
                      class="tooltip-trigger"
                    >
                      
                      <div
                        class="tags has-addons"
                      >
                        <span
                          class="is-series tag"
                          data-v-c1fd21ef=""
                        >
                          
                          3rd
                          
                        </span>
                        <span
                          class="tag"
                        >
                          8
                        </span>
                      </div>
                      
                    </div>
                  </div>
                </div>
                
              </div>
            </td>
            
            <!--v-if-->
          </tr>
          <transition-stub>
            <!--v-if-->
          </transition-stub>
          <!--v-if-->
          
          
          <tr
            class=""
            draggable="false"
          >
            <!--v-if-->
            <!--v-if-->
            
            <td
              class=""
              data-label="PlayStyle"
              scoped=""
            >
              COUPLE
            </td>
            <td
              class=""
              data-label="Difficulty"
              scoped=""
            >
              1
            </td>
            <td
              class=""
              data-label="Level"
              scoped=""
            >
              <div
                class="field is-grouped is-grouped-multiline"
              >
                
                <div
                  class="control"
                >
                  <div
                    class="b-tooltip is-primary is-top"
                  >
                    <transition-stub>
                      <div
                        class="tooltip-content is-top"
                        style="display: none;"
                      >
                        <span
                          class=""
                        />
                        
                        Dance Dance Revolution
                        
                      </div>
                    </transition-stub>
                    <div
                      class="tooltip-trigger"
                    >
                      
                      <div
                        class="tags has-addons"
                      >
                        <span
                          class="is-series tag"
                          data-v-c1fd21ef=""
                        >
                          
                          1st-jp
                          
                        </span>
                        <span
                          class="tag"
                        >
                          6
                        </span>
                      </div>
                      
                    </div>
                  </div>
                </div>
                
              </div>
            </td>
            
            <!--v-if-->
          </tr>
          <transition-stub>
            <!--v-if-->
          </transition-stub>
          <!--v-if-->
          
          
          <tr
            class=""
            draggable="false"
          >
            <!--v-if-->
            <!--v-if-->
            
            <td
              class=""
              data-label="PlayStyle"
              scoped=""
            >
              COUPLE
            </td>
            <td
              class=""
              data-label="Difficulty"
              scoped=""
            >
              2
            </td>
            <td
              class=""
              data-label="Level"
              scoped=""
            >
              <div
                class="field is-grouped is-grouped-multiline"
              >
                
                <div
                  class="control"
                >
                  <div
                    class="b-tooltip is-primary is-top"
                  >
                    <transition-stub>
                      <div
                        class="tooltip-content is-top"
                        style="display: none;"
                      >
                        <span
                          class=""
                        />
                        
                        Dance Dance Revolution
                        
                      </div>
                    </transition-stub>
                    <div
                      class="tooltip-trigger"
                    >
                      
                      <div
                        class="tags has-addons"
                      >
                        <span
                          class="is-series tag"
                          data-v-c1fd21ef=""
                        >
                          
                          1st-jp
                          
                        </span>
                        <span
                          class="tag"
                        >
                          7
                        </span>
                      </div>
                      
                    </div>
                  </div>
                </div>
                
              </div>
            </td>
            
            <!--v-if-->
          </tr>
          <transition-stub>
            <!--v-if-->
          </transition-stub>
          <!--v-if-->
          
          
          <tr
            class=""
            draggable="false"
          >
            <!--v-if-->
            <!--v-if-->
            
            <td
              class=""
              data-label="PlayStyle"
              scoped=""
            >
              COUPLE
            </td>
            <td
              class=""
              data-label="Difficulty"
              scoped=""
            >
              3
            </td>
            <td
              class=""
              data-label="Level"
              scoped=""
            >
              <div
                class="field is-grouped is-grouped-multiline"
              >
                
                <div
                  class="control"
                >
                  <div
                    class="b-tooltip is-primary is-top"
                  >
                    <transition-stub>
                      <div
                        class="tooltip-content is-top"
                        style="display: none;"
                      >
                        <span
                          class=""
                        />
                        
                        Dance Dance Revolution
                        
                      </div>
                    </transition-stub>
                    <div
                      class="tooltip-trigger"
                    >
                      
                      <div
                        class="tags has-addons"
                      >
                        <span
                          class="is-series tag"
                          data-v-c1fd21ef=""
                        >
                          
                          1st-jp
                          
                        </span>
                        <span
                          class="tag"
                        >
                          8
                        </span>
                      </div>
                      
                    </div>
                  </div>
                </div>
                
              </div>
            </td>
            
            <!--v-if-->
          </tr>
          <transition-stub>
            <!--v-if-->
          </transition-stub>
          <!--v-if-->
          
          
          <tr
            class=""
            draggable="false"
          >
            <!--v-if-->
            <!--v-if-->
            
            <td
              class=""
              data-label="PlayStyle"
              scoped=""
            >
              UNISON
            </td>
            <td
              class=""
              data-label="Difficulty"
              scoped=""
            >
              1
            </td>
            <td
              class=""
              data-label="Level"
              scoped=""
            >
              <div
                class="field is-grouped is-grouped-multiline"
              >
                
                <div
                  class="control"
                >
                  <div
                    class="b-tooltip is-primary is-top"
                  >
                    <transition-stub>
                      <div
                        class="tooltip-content is-top"
                        style="display: none;"
                      >
                        <span
                          class=""
                        />
                        
                        Dance Dance Revolution 3rd MIX
                        
                      </div>
                    </transition-stub>
                    <div
                      class="tooltip-trigger"
                    >
                      
                      <div
                        class="tags has-addons"
                      >
                        <span
                          class="is-series tag"
                          data-v-c1fd21ef=""
                        >
                          
                          3rd
                          
                        </span>
                        <span
                          class="tag"
                        >
                          6
                        </span>
                      </div>
                      
                    </div>
                  </div>
                </div>
                
              </div>
            </td>
            
            <!--v-if-->
          </tr>
          <transition-stub>
            <!--v-if-->
          </transition-stub>
          <!--v-if-->
          
          
          <tr
            class=""
            draggable="false"
          >
            <!--v-if-->
            <!--v-if-->
            
            <td
              class=""
              data-label="PlayStyle"
              scoped=""
            >
              UNISON
            </td>
            <td
              class=""
              data-label="Difficulty"
              scoped=""
            >
              2
            </td>
            <td
              class=""
              data-label="Level"
              scoped=""
            >
              <div
                class="field is-grouped is-grouped-multiline"
              >
                
                <div
                  class="control"
                >
                  <div
                    class="b-tooltip is-primary is-top"
                  >
                    <transition-stub>
                      <div
                        class="tooltip-content is-top"
                        style="display: none;"
                      >
                        <span
                          class=""
                        />
                        
                        Dance Dance Revolution 3rd MIX
                        
                      </div>
                    </transition-stub>
                    <div
                      class="tooltip-trigger"
                    >
                      
                      <div
                        class="tags has-addons"
                      >
                        <span
                          class="is-series tag"
                          data-v-c1fd21ef=""
                        >
                          
                          3rd
                          
                        </span>
                        <span
                          class="tag"
                        >
                          7
                        </span>
                      </div>
                      
                    </div>
                  </div>
                </div>
                
              </div>
            </td>
            
            <!--v-if-->
          </tr>
          <transition-stub>
            <!--v-if-->
          </transition-stub>
          <!--v-if-->
          
          
          <tr
            class=""
            draggable="false"
          >
            <!--v-if-->
            <!--v-if-->
            
            <td
              class=""
              data-label="PlayStyle"
              scoped=""
            >
              UNISON
            </td>
            <td
              class=""
              data-label="Difficulty"
              scoped=""
            >
              3
            </td>
            <td
              class=""
              data-label="Level"
              scoped=""
            >
              <div
                class="field is-grouped is-grouped-multiline"
              >
                
                <div
                  class="control"
                >
                  <div
                    class="b-tooltip is-primary is-top"
                  >
                    <transition-stub>
                      <div
                        class="tooltip-content is-top"
                        style="display: none;"
                      >
                        <span
                          class=""
                        />
                        
                        Dance Dance Revolution 3rd MIX
                        
                      </div>
                    </transition-stub>
                    <div
                      class="tooltip-trigger"
                    >
                      
                      <div
                        class="tags has-addons"
                      >
                        <span
                          class="is-series tag"
                          data-v-c1fd21ef=""
                        >
                          
                          3rd
                          
                        </span>
                        <span
                          class="tag"
                        >
                          8
                        </span>
                      </div>
                      
                    </div>
                  </div>
                </div>
                
              </div>
            </td>
            
            <!--v-if-->
          </tr>
          <transition-stub>
            <!--v-if-->
          </transition-stub>
          <!--v-if-->
          
          
          <!--v-if-->
        </tbody>
        <!--v-if-->
      </table>
      
      <transition-stub>
        <!--v-if-->
      </transition-stub>
      
    </div>
    <!--v-if-->
  </div>
</section>
`;
